WordPress/src/main/java/org/wordpress/android/ui/reader/utils/ReaderUtils.kt

```diff
@@ -0,0 +1,605 @@
+     */
+    @JvmStatic
+    fun sanitizeWithDashes(title: String): String {
+        val trimmedTitle = title.trim()
+        return if (isValidUrlEncodedString(trimmedTitle)
+        ) {
+            trimmedTitle
+        } else {
+            trimmedTitle
+                .replace("&[^\\s]*;".toRegex(), "") // remove html entities
+                .replace("[\\.\\s]+".toRegex(), "-") // replace periods and whitespace with a dash
+                .replace(
+                    "[^\\p{L}\\p{Nd}\\-]+".toRegex(),
+                    ""
+                ) // remove remaining non-alphanumeric/non-dash chars (Unicode aware)
+                .replace("--".toRegex(), "-") // reduce double dashes potentially added above
+        }
+    }
+
+    @Suppress("SwallowedException")
+    private fun isValidUrlEncodedString(title: String): Boolean {
+        try {
+            URI.create(title)
+            return true
+        } catch (e: IllegalArgumentException) {
+            return false
+        }
+    }
+
+    /*
+     * returns the long text to use for a like label ("Liked by 3 people", etc.)
+     */ @JvmStatic
+    fun getLongLikeLabelText(
+        context: Context,
+        numLikes: Int,
+        isLikedByCurrentUser: Boolean
+    ): String {
+        if (isLikedByCurrentUser) {
+            return when (numLikes) {
+                1 -> context.getString(R.string.reader_likes_only_you)
+                2 -> context.getString(R.string.reader_likes_you_and_one)
+                else -> {
+                    val youAndMultiLikes = context.getString(R.string.reader_likes_you_and_multi)
+                    String.format(youAndMultiLikes, numLikes - 1)
+                }
+            }
+        } else {
+            return if (numLikes == 1) {
+                context.getString(R.string.reader_likes_one)
+            } else {
+                val likes = context.getString(R.string.reader_likes_multi)
+                String.format(likes, numLikes)
+            }
+        }
+    }
+
+    /*
+     * short like text ("1 like," "5 likes," etc.)
+     */
+    @JvmStatic
+    fun getShortLikeLabelText(context: Context, numLikes: Int): String {
+        return when (numLikes) {
+            0 -> context.getString(R.string.reader_short_like_count_none)
+            1 -> context.getString(R.string.reader_short_like_count_one)
+            else -> {
+                val count = FormatUtils.formatInt(numLikes)
+                String.format(
+                    context.getString(R.string.reader_short_like_count_multi),
+                    count
+                )
+            }
+        }
+    }
+
+    fun getShortCommentLabelText(context: Context, numComments: Int): String {
+        if (numComments == 1) {
+            return context.getString(R.string.reader_short_comment_count_one)
+        }
+        val count = FormatUtils.formatInt(numComments)
+        return String.format(context.getString(R.string.reader_short_comment_count_multi), count)
```

Comment:
```suggestion
        return if (numComments == 1) {
           context.getString(R.string.reader_short_comment_count_one)
        } else {
           val count = FormatUtils.formatInt(numComments)
           String.format(context.getString(R.string.reader_short_comment_count_multi), count)
        }
```

⛏️ A very minor nitpick about the syntax